### PR TITLE
Add `Style/RequireOrder` cop

### PR DESCRIPTION
**Pull request by r7kamura**
Many projects have `require` and `require_relative` in alphabetical order by convention, the rubocop gem itself being one example. I thought it would be good to have a cop to provide such consistency.

As a good example, rubocop-rspec inspects this with tests. Such checks are replaced by this cop.

- https://github.com/rubocop/rubocop-rspec/blob/v2.15.0/spec/project/project_requires_spec.rb

Like `Bundler/OrderedGems`, code separated by a blank line is considered a different group, so if there are obvious dependencies but you really don't want to write `require_relative` in each file, it is better to separate the groups. lib/rubocop/formatter.rb is a good example of that.

- https://github.com/rubocop/rubocop/blob/v1.39.0/lib/rubocop/formatter.rb#L9-L27

Unfortunately, the dependencies written in lib/rubocop.rb are too complex, so I gave up on organizing this at the time of introducing this cop and decided to ignore it from this cop. Personally, if one file A depends on another file B, I prefer to indicate that fact by writing `require_relative` in file A, but I don't think that is the practice in the rubocop gem, so I don't want to go into this issue here.

Since this cop enforces the rule "always sort them in alphabetical order within a group", I don't think false-positive will occur, but applying autocorrection obviously changes the behavior from the original code, so I have set `SafeAutoCorrect: false`.

Ideally, I believe it is a good manner that the code to always be written in such a way that it does not depend on the order of `require`, but I set `Enabled: false` because this cop has a large impact and may or may not be acceptable depending on the preference of the project. I think this cop is rather opinionated, so I would be glad to get your opinion on the policy.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
I love this suggestion and the thoughtful analysis you've submitted alongside it! I'm only wondering if this cop should also suggest to keep `require` and `require_relative` sections ordered as well (e.g. `require` first, `require_relative` afterwards or vice-versa).

File: lib/rubocop/cop/style/require_order.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      #
+      #   # bad
+      #   require_relative 'b'
+      #   require_relative 'a'
+      #
+      #   # good
+      #   require_relative 'a'
+      #   require_relative 'b'
+      #
+      #   # good
```
Comment (bbatsov) [inline lib/rubocop/cop/style/require_order.rb:28]: You might mention here that a blank line introduces a new section or something like this.

**Comment by r7kamura** [comment]
> 'm only wondering if this cop should also suggest to keep `require` and `require_relative` sections ordered as well (e.g. `require` first, `require_relative` afterwards or vice-versa).

I guess it would be difficult to enforce that, since there would be case where you want to write `relative_relative` before `require`, or vice versa, depending on the situation. If we want to do that, it might be better to prepare something like `Style/RequireSectionOrder` separately.

**Comment by koic** [comment]
@zhuravel Thank you for the feedback. That issue will be resolved by #11254.

**Pull request merged**